File: data/multilang_comparison/program_c/src/func5.c
#include "program_c.h" 
array_t* func5(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array121;
   if (pCounter > 0) {
      array121 = vars->data[--pCounter];
      array121->refC++;
      DEBUG_COPY(array121->id);
   } else {
      array121 = (array_t*)malloc(sizeof(array_t));
      array121->size = 506;
      array121->refC = 1;
      array121->id = 121;
      array121->data = (unsigned int*)malloc(array121->size*sizeof(unsigned int));
      memset(array121->data, 0, array121->size*sizeof(unsigned int));
      DEBUG_NEW(array121->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array121;
   array_t* array122 = func10(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array122->id);
   free(params0.data);
   unsigned int loop36 = 0;
   unsigned int loopLimit36 = (50)/3 + 1;
   for(; loop36 < loopLimit36; loop36++) {
      for (int i = 0; i < array121->size; i++) {
         array121->data[i]++;
      }
   }
   for (int i = 0; i < array121->size; i++) {
      if (array121->data[i] == 28) { 
         return array121;
      }
   }
   array121->refC--;
   if(array121->refC == 0) {
      free(array121->data);
      free(array121);
      DEBUG_FREE(array121->id);
   }
   return array122;
}

File: data/multilang_comparison/program_nim/src/func5.nim
proc func5*(vars: ptr ArrayTParam, loopsFactor: cint): ptr ArrayT =
   var pCounter = vars.size
   var array217: ptr ArrayT
   if pCounter > 0:
      dec pCounter
      array217 = vars.data[pCounter]
      inc array217.refC
      debugCopy(array217.id)
   else:
      array217 = cast[ptr ArrayT](alloc(sizeof(ArrayT)))
      array217.size = 757
      array217.refC = 1
      array217.id = 217
      array217.data = cast[ptr UncheckedArray[uint32]](alloc(array217.size * sizeof(uint32)))
      zeroMem(array217.data, array217.size * sizeof(uint32))
      debugNew(array217.id)
   var params0: ArrayTParam
   params0.size = 1
   params0.data = cast[ptr UncheckedArray[ptr ArrayT]](alloc(params0.size * sizeof(ptr ArrayT)))
   params0.data[0] = array217
   var array218: ptr ArrayT = func10(addr params0, getPath(), loopsFactor)
   debugReturn(array218.id)
   dealloc(params0.data)
   let loopLimit63 = (50) div (3) + 1
   for loop63 in 0 ..< loopLimit63:
      discard
      for i in 0 ..< array218.size:
         inc array218.data[i]
   for i in 0 ..< array217.size:
      if array217.data[i] == 59:
         return array217
   dec array218.refC
   if array218.refC == 0:
      dealloc(array218.data)
      dealloc(array218)
      debugFree(array218.id)
   return array217

